esb/plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/debugger/launch/tabs/DebuggerConfigTab.java

```diff
@@ -0,0 +1,182 @@
+		configuration.setAttribute(Messages.DebuggerConfigTab_CommandPortLabel,
+				DEFAULT_COMMAND_PORT);
+		configuration.setAttribute(Messages.DebuggerConfigTab_EventPortLabel,
+				DEFAULT_EVENT_PORT);
+	}
+
+	/**
+	 * Initializes this tab's controls with values from the given launch
+	 * configuration.
+	 * <p>
+	 * This method is called when a configuration is selected to view or edit,
+	 * after this tab's control has been created.
+	 */
+	@Override
+	public void initializeFrom(ILaunchConfiguration launchConfig) {
+		if (StringUtils.isEmpty(commandPort.getText())) {
+			commandPort.setText(DEFAULT_COMMAND_PORT);
+		}
+		if (StringUtils.isEmpty(eventPort.getText())) {
+			eventPort.setText(DEFAULT_EVENT_PORT);
+		}
+	}
+
+	/**
+	 * Copies values from this tab into the given launch configuration.
+	 */
+	@Override
+	public void performApply(ILaunchConfigurationWorkingCopy configuration) {
+
+		configuration.setAttribute(Messages.DebuggerConfigTab_CommandPortLabel,
+				commandPort.getText().toString());
+		configuration.setAttribute(Messages.DebuggerConfigTab_EventPortLabel,
+				eventPort.getText().toString());
+	}
+
+	/**
+	 * Returns whether this tab is in a valid state in the context of the
+	 * specified launch configuration.
+	 * <p>
+	 * This information is typically used by the launch configuration dialog to
+	 * decide when it is okay to launch.
+	 * 
+	 * @return boolean Whether this tab is in a valid state
+	 */
+	@Override
+	public boolean isValid(ILaunchConfiguration launchConfig) {
+
+		try {
+			Integer.parseInt(commandPort.getText());
+			Integer.parseInt(eventPort.getText());
+			setErrorMessage(null);
+			return true;
+		} catch (NumberFormatException e) {
+			setErrorMessage(Messages.DebuggerConfigTab_IntegerPortValueValidationErrorMessage);
+			return false;
+		}
+
+	}
+
+	/**
+	 * @return whether this tab is in a state that allows the launch
+	 *         configuration whose values this tab is showing to be saved.
+	 */
+	@Override
+	public boolean canSave() {
+		return isDirty();
+	}
+
+	/**
+	 * @return the message for this tab which gives information to user.
+	 */
+	@Override
+	public String getMessage() {
+		return Messages.DebuggerConfigTab_MainMessage;
+	}
+
+	/**
+	 * @return the name of this tab.
+	 */
+	@Override
+	public String getName() {
+		return Messages.DebuggerConfigTab_MainTitle;
+	}
+
+	/**
+	 * Creates the top level control for the launch configuration tab under the
+	 * given parent composite.
+	 * 
+	 * <p>
+	 * This method is called once on tab creation, after
+	 * setLaunchConfigurationDialog is called.
+	 */
+	@Override
+	public void createControl(Composite parent) {
+		Composite topControl = new Composite(parent, SWT.NONE);
+		topControl.setLayout(new GridLayout(1, false));
+
+		Group mainGroup = new Group(topControl, SWT.NONE);
+		mainGroup.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false,
+				1, 1));
+		mainGroup.setText(Messages.DebuggerConfigTab_CommandPortLabel); //$NON-NLS-1$
```

Comment:
It silence a warning that Eclipse emits when it encounters a non externalized string, and that it doesn't require localization.